core/src/files.rs

```diff
@@ -0,0 +1,212 @@
+use std::{
+    ffi::{OsStr, OsString},
+    path::PathBuf,
+    rc::Rc,
+};
+
+use codespan::ByteIndex;
+use codespan_reporting::files::Error;
+use nickel_lang_vector::Vector;
+
+use crate::{position::RawSpan, stdlib::StdlibModule};
+
+#[derive(
+    Copy, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, serde::Serialize, serde::Deserialize,
+)]
+pub struct FileId(u32);
```

Comment:
I think the original implementation used `NonZeroU32` (I don't remember the exact syntax) to enable some niche optimization (like `Option<FileId>` taking the same space as `FileId`). It probably doesn't matter on x64, and maybe we don't care at all; just FYI.

Reply:
Right, I noticed that but ignored it because I think we don't store `Option<FileId>` anywhere (we have `TermPos::None` instead)